src/Native/System.Native/pal_mount.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_mount.h"
+#include <assert.h>
+#include <string.h>
+#include <errno.h>
+
+// Check if we should use getmntinfo or /proc/mounts
+#if HAVE_MNTINFO
+#include <sys/mount.h>
+#else
+#include <sys/statfs.h>
+#include <mntent.h>
+#define STRING_BUFFER_SIZE 8192
+#endif
+
+#if HAVE_MNTINFO
+
+int GetMountInfo(MountPointFound onFound)
+{
+    // getmntinfo returns pointers to OS-internal structs, so we don't need to worry about free'ing the object
```

Comment:
The man page for getmntinfo is scaring me:

> The getmntinfo() function writes the array of structures to an internal static object and returns a pointer to that object.  Subsequent calls to getmntinfo() will modify the same object.

No mention of whether said object is at least thread-local. And I don't see an _r version. Perhaps test in parallel while inserting and removing removable drives on OS X? We might need to lock on the managed side.